test/functions/hilde/defaultOf/precedence.chpl

```diff
@@ -19,7 +19,7 @@ proc _defaultOf(type t) where t == R
   return result;
 }
 
-proc R.R()
+proc R.init()
```

Comment:
I proposed that in my original commit, but decided against it as a means of ensuring that "_defaultOf()" doesn't have any (visible) effect in the initializers world anymore.  I think of this as an example of whitebox testing -- something you'd never write if you just read the (final) language definition (i.e., "What if I had a function called _defaultOf()?" is a silly question in such a world), but something worth testing against knowing that the compiler does have special code for how it treats "defaultOf()" in it today.

I decided this before I ran into a related problem in which I was converting `channel` in the standard modules over to initializers, yet the compiler gave an error relating to _defaultOf() functions even though (as far as I could tell) there were no such functions in the module code.  This emphasized for me that _defaultOf() may still play some role in the initializers world and so we should make sure that those impacts aren't user-visible (I think @noakesmichael mentioned that he ran into a _defaultOf()-related problem in initializers code recently also when trying to fix a BHarsh bug?).

I'd be comfortable removing this test at the point when a `git grep defaultOf` had no hits outside of the `test/` directory.